cornell-dti/office-hours
# Updates on average wait time

### Summary 



This PR updates the way we calculate average wait time. Currently, QMI shows the same average wait time for all students regardless of their position in the queue, and students' wait times are considered to have the same weight when we calculate the total wait time of the session. I've changed the way we calculate the average wait time based on this [detailed solution](https://docs.google.com/document/d/1KJhXQdAfdDP_7PQ_5QxpV46ShoCxPvJ4yhZr2ICBWbo/edit?usp=sharing). The major changes are: 

- Added a new optional field called `position` under `FireQuestion` interface 
- Changed `waitTimeChange` to take into account the student's position in the queue 
- Changed estimated wait time to take into account the student's position in the queue by calculating the dynamic position. If the student does not have their question in the queue, their position is considered as _1 + last student's position_. TAs will see the average wait time of the entire session. 






### Test Plan 

- Try adding multiple questions to a session and check that students have different average wait times based on their position. 
- Check that the new `position` field does not cause an error



### Notes 



### Breaking Changes 

None
